core/src/main/java/org/mskcc/cbio/portal/servlet/PathologyReportView.java

```diff
@@ -0,0 +1,224 @@
+import org.mskcc.cbio.portal.dao.DaoPatient;
+import org.mskcc.cbio.portal.dao.DaoSample;
+import org.mskcc.cbio.portal.model.CancerStudy;
+import org.mskcc.cbio.portal.model.Patient;
+import org.mskcc.cbio.portal.model.Sample;
+import org.mskcc.cbio.portal.util.AccessControl;
+import org.mskcc.cbio.portal.util.GlobalProperties;
+import org.mskcc.cbio.portal.util.SpringUtil;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.net.URLDecoder;
+import org.apache.commons.io.FilenameUtils;
+
+/**
+ * A servlet to respond to requests for Pathology Report files for cancer studies.
+ * Adapted from CancerStudyView.java
+ * 
+ * @author Shaun Muscat
+ */
+public class PathologyReportView extends HttpServlet {
+    private static Logger logger = Logger.getLogger(PathologyReportView.class);
+    public static final String ERROR_CODE = "error_code";
+    public static final String ERROR_MSG = "error_msg";
+
+    // class which process access control to cancer studies
+    private AccessControl accessControl;
+
+    // root directory of internal pathology reports or null if undefined
+    private static final String DATA_DIRECTORY = GlobalProperties.getInternalPathReportRoot();
+    
+    /**
+     * Initializes the servlet.
+     *
+     * @throws ServletException Serlvet Init Error.
+     */
+    @Override
+    public void init() throws ServletException {
+        super.init();
+		accessControl = SpringUtil.getAccessControl();
+    }
+    
+    /** 
+     * Processes requests for both HTTP <code>GET</code> and <code>POST</code> methods.
+     * @param request servlet request
+     * @param response servlet response
+     * @throws ServletException if a servlet-specific error occurs
+     * @throws IOException if an I/O error occurs
+     */
+    protected void processRequest(HttpServletRequest request, HttpServletResponse response)
+        throws ServletException, IOException {        
+        try {            
+            validateRequest(request);
+            if (request.getAttribute(ERROR_CODE)!=null) {
+                response.sendError(Integer.parseInt((String) request.getAttribute(ERROR_CODE)), 
+                    (String) request.getAttribute(ERROR_MSG));
+            } else {
+                File pathologyReport = getRequestedFile(request);
+                serveFile(response, pathologyReport);
+            }
+        } catch (DaoException e) {
+            logger.error("Got Database Exception while processing request for pathology report.", e);
+            response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, 
+                "An error occurred while trying to connect to the database.");
+        } catch (NoSuchFileException e) {
```

Comment:
I've had a look and this can be taken out anyway, as it doesn't actually get thrown by anything. All of the `File` and `Files` class methods used in here only seem to throw `IO exception`.